mount/mount_windows.go

```diff
@@ -38,18 +38,22 @@ var (
 	ErrNotImplementOnWindows = errors.New("not implemented under windows")
 )
 
-// Mount to the provided target.
-func (m *Mount) mount(target string) (retErr error) {
-	readOnly := false
+func (m *Mount) ReadOnly() bool {
 	for _, option := range m.Options {
 		if option == "ro" {
-			readOnly = true
-			break
+			return true
 		}
 	}
+	return false
+}
 
+// Mount to the provided target.
+func (m *Mount) mount(target string) (retErr error) {
 	if m.Type == "bind" {
-		if err := bindfilter.ApplyFileBinding(target, m.Source, readOnly); err != nil {
+		if !strings.EqualFold(filepath.Base(m.Source), "Files") {
+			m.Source = filepath.Join(m.Source, "Files")
+		}
+		if err := bindfilter.ApplyFileBinding(target, m.Source, m.ReadOnly()); err != nil {
```

Comment:
I'd prefer to not modify `m` in-place (which I assume is why we check for a second `Files`), partly because this function really has no place modifying `m`, and partly because _this package_ should not be assuming that some person hasn't unexpectedly created a base layer in a directory named `Files`.

That would do the wrong thing with this check; the built-in Windows snapshotter will never do that, but an alternative implementation in an external snapshotter plugin might?

So I expect that
```
if err := bindfilter.ApplyFileBinding(target, filepath.Join(m.Source, "Files"), m.ReadOnly()); err != nil {
```
would work fine, unless I'm overlooking something.

Reply:
Okay. If I understand this correctly, in the case of bind mounts we need to:

  * Mount the layer's ```Files``` folder in the target
  * Make sure that ```Files``` folder actually exists. This should happen in ```hcsshim``` with the new base layer code. It should be fine not to check for the existence of that folder.

Reply:
made the change